client/src/js/controller.js

```diff
@@ -61,6 +67,28 @@ module.exports = function (app) {
             }
           });
         };
+        var socialClassToUrl = {
+          'twitter': 'https://twitter.com/intent/tweet?original_referer=http%3A%2F%2Fwww.gapminder.org&amp;related=Gapminder&amp;text=Gapminder&amp;tw_p=tweetbutton&amp;url=',
+          'facebook': 'https://www.facebook.com/sharer/sharer.php?u=',
+          'mail': 'mailto:?subject=Gapminder&body='
+        };
+
+        var arrayClass = Object.keys(socialClassToUrl);
+
+        function eventListenerShareLink() {
+          var params = getDocumentUrl();
+          getJSON(address, params, function (response) {
+            if (response.status_code === 200) {
+              arrayClass.forEach(function (className) {
+                Array.prototype.forEach.call(document.querySelectorAll('.' + className), function (node) {
```

Comment:
`document.querySelectorAll('.' + className)`

This should be queried only once and cached in order to avoid tree walking for searching target elements